File: 백준/Silver/11723. 집합/집합.py
import sys
input = sys.stdin.readline

n = int(input())
arr = set()
for _ in range(n):
    lst = list(input().split())
    a = lst[0]
    if a=='add':
        if int(lst[1]) not in arr:
            arr.add(int(lst[1]))
    elif a=='remove':
        if int(lst[1]) in arr:
            arr.remove(int(lst[1]))
    elif a=='check':
        if int(lst[1]) in arr:
            print(1)
        else:
            print(0)
    elif a=='toggle':
        if int(lst[1]) in arr:
            arr.remove(int(lst[1]))
        else:
            arr.add(int(lst[1]))
    elif a=='all':
        arr = set([i for i in range(1, 21)])
    elif a=='empty':
        arr = set()

File: 백준/Silver/11723. 집합/집합.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int m = Integer.parseInt(br.readLine());
        boolean[] arr = new boolean[21];
        for (int i = 0; i<m; i++) {
            String a = br.readLine();
            String[] b = a.split(" ");
            if (b[0].equals("add")) {
                if (!arr[Integer.parseInt(b[1])])  {
                    arr[Integer.parseInt(b[1])] = true;
                }
            } else if (b[0].equals("remove")) {
                if (arr[Integer.parseInt(b[1])]) {
                    arr[Integer.parseInt(b[1])] = false;
                }
            } else if (b[0].equals("check")) {
                if (arr[Integer.parseInt(b[1])]) {
                    sb.append(1 + "\n");
                } else {
                    sb.append(0 + "\n");
                }
            } else if (b[0].equals("toggle")) {
                if (arr[Integer.parseInt(b[1])]) {
                    arr[Integer.parseInt(b[1])] = false;
                } else {
                    arr[Integer.parseInt(b[1])] = true;
                }
            } else if (b[0].equals("all")) {
                for (int j = 1; j<=20; j++) {
                    arr[j] = true;
                }
            } else if (b[0].equals("empty")) {
                for (int j = 1; j<=20; j++) {
                    arr[j] = false;
                }
            }

        }
        System.out.println(sb);
    }
}